sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -328,6 +335,33 @@ private [sql] case class CreateTempTableUsingAsSelect(
   }
 }
 
+private[sql] case class DropTable(
+    tableName: String,
+    isExists: Boolean,
+    temporary: Boolean) extends Command
+
+private[sql] case class DropTempTable(
+    tableIdent: Seq[String],
+    isExists: Boolean,
+    temporary: Boolean) extends RunnableCommand {
+
+  def run(sqlContext: SQLContext) = {
+    val tableExists = sqlContext.catalog.tableExists(tableIdent)
+    val tableName = tableIdent.head
+    if (isExists) {
+      if (tableExists) sqlContext.dropTempTable(tableName)
+      else logWarning(s"Unknown table '$tableName'")
+    }
+    // no `IF EXISTS` keyword, will thorw an exception
+    else {
+      if (tableExists) sqlContext.dropTempTable(tableName)
+      else sys.error(s"Unknown table '$tableName'")
+    }
+    Seq.empty
```

Comment:
I'm not sure this logic is exactly right.  I think we need to check both if it exists and if it is a temporary table.